### DISCUSSION TITLE
earlyaccess behavior?

### DISCUSSION THREAD

**Discussion by maxandersen** [jreleaser/jreleaser#1565] [Q&A]
Hi,

I'm trying to grok why my setup at https://github.com/maxandersen/hass-cli creates earlyaccess releases that makes a tag in my repo. Even when I'm doing a full release.

I tried looking in the docs for description on how earlyaccess releases happens but thus far I'm stumped.

I'm obviously missing something but can't spot what is different in my setup and i.e. https://github.com/jreleaser/helloworld-java-jar/blob/main/.github/workflows/release.yml

Except that I don't call out explicitly full-release...but is that not the default action anyway?

anyhow - looking to find info on whywhen I do a release with version set to ie. 0.2.0 that jreleaser decides to call it earlyaccess...seems weird to me ;)

**Comment by aalmiray** [comment]
The default behavior is to detect if a release is an earlyaccess one based on the project version. If it ends with `-SNAPSHOT` then it's deemed as such. The choice of release command does not affect the outcome in this regard, rather it's the configuration itself. 

The snapshot pattern may be changed to something else of course.

**Reply by maxandersen** [reply]
Hmm…but I set and committed the version explicitly to be non snapshot. I guess I’ll try again.

**Reply by maxandersen** [reply]
in anycase - seems to work now as my third attempt had 0.0.3 show up at https://github.com/maxandersen/home-assistant-qli/releases.

Still two questions in my mind:

1) is the earlyaccess behavior documented somewhere ? I don't find it in the docs.

2) changelog on earlyaccess only show last commits since last earlyaccess...should it not show changelog since the actual last release ?

**Comment by aalmiray** [comment]
>     1. is the earlyaccess behavior documented somewhere ? I don't find it in the docs.

This is hinted in the examples but not it's not explicitly discussed. Filed https://github.com/jreleaser/jreleaser.github.io/issues/70

>     2. changelog on earlyaccess only show last commits since last earlyaccess...should it not show changelog since the actual last release ?

This is the default behavior for snapshot releases. Changelog is calculated from the latest `early-access` tag or latest releases, whichever is newer. However, you can alter this behavior by setting `project.snapshot.fullChangelog` to `true`. Please see the options available at https://jreleaser.org/guide/latest/reference/project.html
